x-pack/plugin/slm/src/internalClusterTest/java/org/elasticsearch/xpack/slm/SLMHealthBlockedSnapshotIT.java

```diff
@@ -0,0 +1,343 @@
+        // allocate index to the specified data node
+        Settings settings = indexSettings(1, 0).put("index.routing.allocation.require._name", dataNodeName).build();
+        createIndex(idxName, settings);
+
+        logger.info("--> indexing some data");
+        final int numdocs = randomIntBetween(10, 100);
+        IndexRequestBuilder[] builders = new IndexRequestBuilder[numdocs];
+        for (int i = 0; i < builders.length; i++) {
+            builders[i] = prepareIndex(idxName).setId(Integer.toString(i)).setSource("field1", "bar " + i);
+        }
+        indexRandom(true, builders);
+        indicesAdmin().refresh(new RefreshRequest(idxName)).actionGet();
+    }
+
+    private void putSnapshotPolicy(
+        String policyName,
+        String snapshotNamePattern,
+        String schedule,
+        String repoId,
+        String indexPattern,
+        TimeValue unhealthyIfNoSnapshotWithin
+    ) {
+        Map<String, Object> snapConfig = new HashMap<>();
+        snapConfig.put("indices", Collections.singletonList(indexPattern));
+        snapConfig.put("ignore_unavailable", false);
+        snapConfig.put("partial", true);
+
+        SnapshotLifecyclePolicy policy = new SnapshotLifecyclePolicy(
+            policyName,
+            snapshotNamePattern,
+            schedule,
+            repoId,
+            snapConfig,
+            SnapshotRetentionConfiguration.EMPTY,
+            unhealthyIfNoSnapshotWithin
+        );
+
+        PutSnapshotLifecycleAction.Request putLifecycle = new PutSnapshotLifecycleAction.Request(
+            TEST_REQUEST_TIMEOUT,
+            TEST_REQUEST_TIMEOUT,
+            policyName,
+            policy
+        );
+        try {
+            client().execute(PutSnapshotLifecycleAction.INSTANCE, putLifecycle).get();
+        } catch (Exception e) {
+            logger.error("failed to create slm policy", e);
+            fail("failed to create policy " + policy + " got: " + e);
+        }
+    }
+
+    private void assertSlmYellowMissingSnapshot(int numUnhealthyPolicy) throws Exception {
+        assertBusy(() -> {
+            GetHealthAction.Request getHealthRequest = new GetHealthAction.Request(true, 1000);
+            GetHealthAction.Response health = admin().cluster().execute(GetHealthAction.INSTANCE, getHealthRequest).get();
+            assertThat(health.getStatus(), equalTo(HealthStatus.YELLOW));
+            HealthIndicatorResult slmIndicator = health.findIndicator(SlmHealthIndicatorService.NAME);
+            assertThat(slmIndicator.status(), equalTo(HealthStatus.YELLOW));
+            assertThat(slmIndicator.impacts().size(), equalTo(1));
+            List<HealthIndicatorImpact> missingSnapshotPolicies = slmIndicator.impacts()
+                .stream()
+                .filter(impact -> SlmHealthIndicatorService.MISSING_SNAPSHOT_IMPACT_ID.equals(impact.id()))
+                .toList();
+            assertThat(missingSnapshotPolicies.size(), equalTo(numUnhealthyPolicy));
```

Comment:
Instead of just asserting the size, should we also assert that the correct policy names are unhealthy? I think that would make it air-tight.